Source/JavaScriptCore/dfg/DFGByteCodeParser.cpp

```diff
@@ -5456,6 +5463,284 @@ void ByteCodeParser::parseBlock(unsigned limit)
         return codeBlock->outOfLineJumpOffset(m_currentInstruction);
     };
 
+    auto resolveScopeHelper = [&](auto& bytecode, Operand dst) {
+        auto& metadata = bytecode.metadata(codeBlock);
+        ResolveType resolveType;
+        unsigned depth;
+        JSScope* constantScope = nullptr;
+        JSCell* lexicalEnvironment = nullptr;
+        SymbolTable* symbolTable = nullptr;
+        {
+            ConcurrentJSLocker locker(m_inlineStackTop->m_profiledBlock->m_lock);
+            resolveType = metadata.m_resolveType;
+            depth = metadata.m_localScopeDepth;
+            switch (resolveType) {
+            case GlobalProperty:
+            case GlobalVar:
+            case GlobalPropertyWithVarInjectionChecks:
+            case GlobalVarWithVarInjectionChecks:
+            case GlobalLexicalVar:
+            case GlobalLexicalVarWithVarInjectionChecks:
+                constantScope = metadata.m_constantScope.get();
+                break;
+            case ModuleVar:
+                lexicalEnvironment = metadata.m_lexicalEnvironment.get();
+                break;
+            case ResolvedClosureVar:
+            case ClosureVar:
+            case ClosureVarWithVarInjectionChecks:
+                symbolTable = metadata.m_symbolTable.get();
+                break;
+            default:
+                break;
+            }
+        }
+
+        if (needsDynamicLookup(resolveType, op_resolve_scope)) {
+            unsigned identifierNumber = m_inlineStackTop->m_identifierRemap[bytecode.m_var];
+            set(dst, addToGraph(ResolveScope, OpInfo(identifierNumber), get(bytecode.m_scope)));
```

Comment:
This doesn't look right. It's not putting the result in `dst`

Reply:
Also, this would be a compile error once we get rid of that `scope` field we no longer need from the bytecode

Reply:
> This doesn't look right. It's not putting the result in `dst`

I think this is correct. Here, the `dst` is passed to the helper function `resolveScopeHelper `, which is extracted from `case op_resolve_scope:` and it is a helper function for both op_resolve_scope and op_resolve_and_get_from_scope. Each case will pass their own `dst`, which is the destination register  or Tmp variable for op_resolve_scope or op_resolve_and_get_from_scope respectively. 

> Also, this would be a compile error once we get rid of that scope field we no longer need from the bytecode

I don't think we can remove the `scope` operand or field.